### PR TITLE
add feature about vllm backend

### PR DESCRIPTION
**Pull request by Jerry-Kon**

### CODE REVIEW

File: examples/chatglm2.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 from llmlite.apis import ChatLLM, ChatMessage
 
 chat = ChatLLM(
-    model_name_or_path="THUDM/chatglm2-6b",
+    model_name_or_path="/workspace/models/thudm/chatglm2-6b",
```
Comment (kerthcet) [inline examples/chatglm2.py:4]: Revert this, this is more general.

File: examples/llama2.py [hunk 2]
```diff
@@ -1,9 +1,9 @@
 from llmlite.apis import ChatLLM, ChatMessage
 
 chat = ChatLLM(
-    model_name_or_path="/data/models/meta-llama/Llama-2-7b-chat-hf",
+    model_name_or_path="/workspace/models/meta-llama/llama-2-7b-chat-hf",
```
Comment (kerthcet) [inline examples/llama2.py:4]:
```suggestion
    model_name_or_path="meta-llama/Llama-2-7b-chat-hf",
```

File: pyproject.toml [hunk 3]
```diff
@@ -9,12 +9,13 @@ readme = "README.md"
...
-transformers = "4.33.1"
-tokenizers = "0.13.3"
+transformers = "4.34.0"
 sentencepiece = "0.1.99"
-torch = "2.0.0"
 pydantic = "<2"
 openai = "^0.28.0"
+vllm = "0.2.3"
+tokenizers = "0.14"
+torch = "^2.1.1"
```
Comment (kerthcet) [inline pyproject.toml:18]: Have you tested with this version? Can you run the model with vllm normally?

File: examples/chatglm2.py [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
     task="text-generation",
+    backend="vllm",
```
Comment (kerthcet) [inline examples/chatglm2.py:6]: comment this line, people can uncomment this if needed.

File: examples/llama2.py [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
     task="text-generation",
-    backend="hf",
+    backend="vllm",
```
Comment (kerthcet) [inline examples/llama2.py:6]: ditto

File: llmlite/backends/vllm_backend.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 from typing import Optional
 
 import torch
+from vllm import LLM as VLLM
```
Comment (kerthcet) [inline llmlite/backends/vllm_backend.py:4]:
```suggestion
from vllm import LLM as vllm
```

**Comment by kerthcet** [comment]
I'll merge this PR first because I want to add more features ASAP, so I'll help you fix the comments. Thanks.

File: pyproject.toml (hunk 3 above)
Comment (Jerry-Kon) [reply pyproject.toml:18]: I have runned the code, by this version.

**Pull request merged**